flying-saucer-core/src/main/java/org/xhtmlrenderer/swing/NaiveUserAgent.java

```diff
@@ -117,21 +120,89 @@ public void clearImageCache() {
      * @return The stylesheet value
      */
     //TOdO:implement this with nio.
-    protected InputStream resolveAndOpenStream(String uri) {
+    protected InputStream resolveAndOpenStream(final String uri) {
         java.io.InputStream is = null;
-        uri = resolveURI(uri);
+        String resolvedUri = resolveURI(uri);
         try {
-            is = new URL(uri).openStream();
+            is = openStream(resolvedUri);
         } catch (java.net.MalformedURLException e) {
-            XRLog.exception("bad URL given: " + uri, e);
+            XRLog.exception("bad URL given: " + resolvedUri, e);
         } catch (java.io.FileNotFoundException e) {
-            XRLog.exception("item at URI " + uri + " not found");
+            XRLog.exception("item at URI " + resolvedUri + " not found");
         } catch (java.io.IOException e) {
-            XRLog.exception("IO problem for " + uri, e);
+            XRLog.exception("IO problem for " + resolvedUri, e);
         }
         return is;
     }
+	
+	protected InputStream openStream(String uri) throws MalformedURLException, IOException {
+		return openConnection(uri).getInputStream();
+	}
 
+	/**
+	 * Opens a connections to uri.
+	 * 
+	 * This can be overwritten to customize handling of connections by type.
+	 * 
+	 * @param uri the uri to connect to
+	 * @return URLConnection opened connection to uri
+	 * @throws IOException if an I/O exception occurs.
+	 */
+	protected URLConnection openConnection(String uri) throws IOException {
+		URLConnection connection = new URL(uri).openConnection();
+		if (connection instanceof HttpURLConnection) {
+			connection = onHttpConnection((HttpURLConnection) connection);
```

Comment:
May be this could be just:

```java
    ((HttpURLConnection) connection).setInstanceFollowRedirects(true);
```

-    https://docs.oracle.com/javase/8/docs/api/java/net/HttpURLConnection.html#setInstanceFollowRedirects-boolean-